spacy/language.py

```diff
@@ -417,11 +680,20 @@ def remove_pipe(self, name):
         if name not in self.pipe_names:
             raise ValueError(Errors.E001.format(name=name, opts=self.pipe_names))
         removed = self.pipeline.pop(self.pipe_names.index(name))
+        # We're only removing the component itself from the metas/configs here
+        # because factory may be used for something else
+        self._pipe_meta.pop(name)
+        self._pipe_configs.pop(name)
```

Comment:
Removing the config is definitely right here, as it's the inverse of what we do in `nlp.add_pipe`.

It's worth noting a small inconsistency here though. In most other places, the config doesn't update when you change the object state. But this does, because we're removing the pipeline from the meta.

What we're doing is right and any alternative would be worse, but the inconsistency is unfortunate. I'm not sure whether users will notice/care, but some might be confused.